File: swift/week-08-project-todo/solutions.swift
import Foundation

// 1. Dictionary mit id (Zufallszahl), text, done: false
func createTodo(text: String) -> [String: Any] {
    fatalError("TODO: Erstelle ein Dictionary mit \"id\" (Zufallszahl), \"text\" (text) und \"done\" (false)")
}

// 2. Todo hinzufügen
func addTodo(todos: [[String: Any]], text: String) -> [[String: Any]] {
    fatalError("TODO: Füge ein neues Todo (mit createTodo) zu todos hinzu und gib die neue Liste zurück")
}

// 3. Nur text-Felder zurückgeben
func listTodos(todos: [[String: Any]]) -> [String] {
    fatalError("TODO: Gehe alle Todos durch und gib ein Array mit allen \"text\"-Werten zurück")
}

// 4. done = true für das Todo mit der gegebenen id
func markDone(todos: [[String: Any]], id: Int) -> [[String: Any]] {
    fatalError("TODO: Finde das Todo mit der passenden id und setze \"done\" auf true")
}

// 5. Todo mit der gegebenen id entfernen
func removeTodo(todos: [[String: Any]], id: Int) -> [[String: Any]] {
    fatalError("TODO: Entferne das Todo mit der passenden id aus der Liste")
}

File: rust/week_08_project_todo/src/solutions.rs
// TODO: Füge die Felder id: u32, text: String und done: bool hinzu
pub struct TodoItem {
    pub id: u32,
    pub text: String,
    pub done: bool,
}

// TODO: Füge ein neues Todo hinzu (id = todos.len() + 1, done = false)
pub fn add_todo(_todos: &mut Vec<TodoItem>, _text: &str) {
    todo!()
}

// TODO: Gib nur die text-Felder als Vec<&str> zurück
pub fn list_todos(_todos: &[TodoItem]) -> Vec<&str> {
    todo!()
}

// TODO: Setze done = true für das Todo mit der gegebenen id
pub fn mark_done(_todos: &mut [TodoItem], _id: u32) {
    todo!()
}

// TODO: Entferne das Todo mit der gegebenen id
pub fn remove_todo(_todos: &mut Vec<TodoItem>, _id: u32) {
    todo!()
}
